spec/reservation_manager_spec.rb

```diff
@@ -0,0 +1,252 @@
+require_relative "spec_helper"
+
+describe "ReservationManager" do
+  let(:reservation_manager) {
+    Hotel::ReservationManager.new
+  }
+  describe "initialize" do
+    it "can be instantiated" do
+      expect(reservation_manager).must_be_kind_of Hotel::ReservationManager
+    end
+
+    it "establishes the base structures when instantiated" do
+      [:rooms, :reservations, :blocks].each do |prop|
+        expect(reservation_manager).must_respond_to prop
+      end
+
+      expect(reservation_manager.rooms).must_be_kind_of Array
+      expect(reservation_manager.reservations).must_be_kind_of Array
+      expect(reservation_manager.blocks).must_be_kind_of Array
+    end
+
+    it "has 20 rooms" do
+      expect(reservation_manager.rooms.length).must_equal 20
+    end
+  end
+
+  describe "reserve method" do
+    let(:new_reservation) {
+      reservation_manager.reserve(room_id: 1,
+                                  check_in_date: "2019-03-12",
+                                  check_out_date: "2019-03-15")
+    }
+    it "can reserve an available room and adds a new reservation to the list of reservations" do
+      before_reserve = reservation_manager.reservations.length
+      new_reservation
+      after_reserve = reservation_manager.reservations.length
+      expect(after_reserve - before_reserve).must_equal 1
+    end
+
+    it "cannot reserve unavailable rooms" do
+      new_reservation
+      expect { reservation_manager.reserve(new_reservation) }.must_raise ArgumentError
+    end
+
+    it "cannot reserve a room set in the block" do
+      reservation_manager.create_block(room_ids: [1, 2, 3],
+                                       check_in_date: "2019-03-10",
+                                       check_out_date: "2019-03-15",
+                                       discount_rate: 0.10)
+      expect {
```

Comment:
You should also have edge-case tests, like reserving a room on the same day a previous reservation ends, and on the same day one begins.  

You should also test reserving two different rooms on the same dates.